### PR TITLE
submodule cloning may not be important to the job, ignore errors

### PR DESCRIPTION
**Pull request by jakecoffman**
In #5982 we enabled submodule cloning by default. However, sometimes a repo has a submodule that is inaccessible to Dependabot that is unrelated to dependencies.

We talked about adding a flag to disable it, but it's nice when Dependabot just works. 

So for this PR I took the approach to just ignore the submodule clone errors and continue with the update. That way if it was  not important, it will succeed in doing the update.

If it was important it will fail later, but I added an error log that says the nature of the failure during the fetch as a hint as to what might be wrong.

### CODE REVIEW

File: common/lib/dependabot/file_fetchers/base.rb [hunk 1]
```diff
@@ -592,19 +595,34 @@ def _clone_repo_contents(target_directory:)
...
+            raise unless e.message.match?(GIT_SUBMODULE_ERROR_REGEX) && e.message.include?("submodule")
+
+            submodule_cloning_failed = true
+            match = e.message.match(GIT_SUBMODULE_ERROR_REGEX)
+            url = match.named_captures["url"]
+            code = match.named_captures["code"]
+
+            # Submodules might be in the repo but unrelated to dependencies,
+            # so ignoring this error to try the update anyway since the base repo exists.
+            Dependabot.logger.error("Cloning of submodule failed: #{url} error: #{code}")
```
Comment (landongrindheim) [inline common/lib/dependabot/file_fetchers/base.rb:616]: 👌 I can picture this being really helpful 😄

File: README.md [hunk 2]
```diff
@@ -271,9 +271,7 @@ classes:
 
 The high-level flow looks like this:
 
-<p align="center">
-  <img src="https://s3.eu-west-2.amazonaws.com/dependabot-images/package-manager-architecture.svg" alt="Dependabot architecture">
-</p>
+![Dependabot high-level architecture diagram](architecture.svg)
```
Comment (landongrindheim) [inline README.md:274]: Was this added intentionally? Should it be a separate commit?

**Review by landongrindheim — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (jakecoffman) [reply README.md:274]: Oops, bad squash I guess. That was something I fixed this morning in another PR.
